e2e/test/scenarios/metrics/metrics-editing.cy.spec.ts

```diff
@@ -98,6 +112,57 @@ describe("scenarios > metrics", () => {
       runQuery();
       verifyScalarValue("18,760");
     });
+
+    it("should be able to rename a metric", () => {
+      const newTitle = "New metric name";
+      createQuestion(ORDERS_COUNT_TABLE_METRIC).then(({ body: card }) => {
+        visitMetric(card.id);
+        renameMetric(newTitle);
+        visitMetric(card.id);
+        queryBuilderHeader().findByDisplayValue(newTitle).should("be.visible");
+      });
+    });
+
+    it("should be able to change the query definition of a metric", () => {
+      createQuestion(ORDERS_COUNT_TABLE_METRIC).then(({ body: card }) =>
+        visitMetric(card.id),
+      );
+      openQuestionActions();
+      popover().findByText("Edit metric definition").click();
+      addBreakout({ tableName: "Product", columnName: "Category" });
+      updateMetric();
+      verifyLineAreaBarChart({ xAxis: "Product → Category", yAxis: "Count" });
+    });
+
+    it("should be able to change the query definition of a metric based on a model", () => {
+      createQuestion(ORDERS_COUNT_MODEL_METRIC).then(({ body: card }) =>
+        visitMetric(card.id),
+      );
+      openQuestionActions();
+      popover().findByText("Edit metric definition").click();
+      addBreakout({ tableName: "Product", columnName: "Category" });
+      updateMetric();
+      verifyLineAreaBarChart({ xAxis: "Product → Category", yAxis: "Count" });
+    });
+
+    it("should pin new metrics automatically", () => {
```

Comment:
collection-based tests would go to a separate file, but I wanted to make sure that the entire flow works, e.g. when you create a metric via UI and not by API